### PR TITLE
Add dep for vendor management

### PR DESCRIPTION
**Pull request by cheapRoc**
I've also updated the `Dockerfile` as appropriate.

### CODE REVIEW

**Comment by cheapRoc** [comment]
This is a prelude to #6 which will use this commit for adding the new dependency on `triton-go`.

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,22 @@
-FROM golang:1.6-alpine
+FROM golang:1.9-alpine
 
-RUN apk add --no-cache gcc git libc-dev
+RUN apk add --no-cache gcc git libc-dev curl
 
-ENV DOCKER_MACHINE_VERSION v0.7.0
+ENV DOCKER_MACHINE_VERSION v0.12.2
+RUN git clone --depth 1 -b "$DOCKER_MACHINE_VERSION" https://github.com/docker/machine.git "$GOPATH/src/github.com/docker/machine" \
+    && go build -v -o "$GOPATH/bin/docker-machine" github.com/docker/machine/cmd
```
Comment (tianon) [inline Dockerfile]: This is coming in via `dep` now, isn't it?  Do we still need to pull it in manually this way?

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+    && mv /tmp/dep /usr/local/bin/dep \
+    && chmod 0755 /usr/local/bin/dep
 
 WORKDIR $GOPATH/src/github.com/joyent/docker-machine-driver-triton
 COPY *.go ./
-
-RUN go install -v ./...
+COPY Gopkg.* ./
+RUN dep ensure \
+    && go install -v ./...
```
Comment (tianon) [inline Dockerfile]: I think we should either re-order this so we don't have to re-download all the deps for every `.go` change, or simply use the `vendor/` directory here too (which is probably better/saner).

So something like this _should_ do the trick:
```dockerfile
WORKDIR $GOPATH/src/github.com/joyent/docker-machine-driver-triton
COPY *.go vendor ./
RUN go install -v ./...
```
(and then we don't need `dep` in the `Dockerfile` at all, unless you want to use this `Dockerfile` for creating a dev environment for folks like me who don't have the `dep` tool installed on their host :smile:)

Comment (cheapRoc) [reply Dockerfile]: Definitely debatable. The thought was to use this as a development environment (for folks that need it) and not simply a container to run `docker-machine`. That interpretation came from the original `Dockerfile` which built everything within the container as well.

But I can remove `dep` and just utilize the `vendor/` directory. This isn't a huge moving target. 👍

File: Dockerfile (hunk 1 above)
Comment (cheapRoc) [reply Dockerfile]: Nice catch, I'll make the change.

**Comment by cheapRoc** [comment]
I'm not a huge fan of `dep` or Go's dependency management either... but the productivity gains outweigh anything else.

**Comment by cheapRoc** [comment]
I kept the bit that installs `docker-machine` within the `Dockerfile` because vendor doesn't include all dependencies of `docker-machine` itself (since the driver doesn't import all of them).

**Pull request merged**